### PR TITLE
[MM-25713] Adds props check to post import validator

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Ticket Link
https://mattermost.atlassian.net/browse/MM-25713

### CODE REVIEW

File: app/import_functions.go [hunk 1]
```diff
@@ -1080,7 +1080,7 @@ func (a *App) importAttachment(data *AttachmentImportData, post *model.Post, tea
 	}
 	fileInfo, appErr := a.DoUploadFile(timestamp, teamId, post.ChannelId, post.UserId, file.Name(), buf.Bytes())
 	if appErr != nil {
-		mlog.Error("Failed to upload file:", mlog.Err(err))
+		mlog.Error("Failed to upload file:", mlog.Err(appErr))
```
Comment (grundleborg) [inline app/import_functions.go:1083]: :facepalm:

File: app/import_validators_test.go [hunk 2]
```diff
@@ -895,109 +895,131 @@ func TestImportValidateReplyImportData(t *testing.T) {
...
+
+	t.Run("Test with missing required properties", func(t *testing.T) {
+		data := PostImportData{
+			Channel:  ptrStr("channelname"),
+			User:     ptrStr("username"),
+			Message:  ptrStr("message"),
+			CreateAt: ptrInt64(model.GetMillis()),
+		}
+		err := validatePostImportData(&data, maxPostSize)
+		require.NotNil(t, err, "Should have failed due to missing required property.")
```
Comment (isacikgoz) [inline app/import_validators_test.go:918]:
```suggestion
		require.NotNil(t, err, "Should have failed due to missing required property.")
		assert.Equal(t, err.Id, "app.import.validate_post_import_data.team_missing.error")
```

File: app/import_validators_test.go [hunk 3]
```diff
@@ -895,109 +895,131 @@ func TestImportValidateReplyImportData(t *testing.T) {
...
+
+		data = PostImportData{
+			Team:     ptrStr("teamname"),
+			User:     ptrStr("username"),
+			Message:  ptrStr("message"),
+			CreateAt: ptrInt64(model.GetMillis()),
+		}
+		err = validatePostImportData(&data, maxPostSize)
+		require.NotNil(t, err, "Should have failed due to missing required property.")
```
Comment (isacikgoz) [inline app/import_validators_test.go:927]:
```suggestion
		require.NotNil(t, err, "Should have failed due to missing required property.")
		assert.Equal(t, err.Id, "app.import.validate_post_import_data.channel_missing.error")
```

File: app/import_validators_test.go [hunk 4]
```diff
@@ -895,109 +895,131 @@ func TestImportValidateReplyImportData(t *testing.T) {
...
+
+		data = PostImportData{
+			Team:     ptrStr("teamname"),
+			Channel:  ptrStr("channelname"),
+			Message:  ptrStr("message"),
+			CreateAt: ptrInt64(model.GetMillis()),
+		}
+		err = validatePostImportData(&data, maxPostSize)
+		require.NotNil(t, err, "Should have failed due to missing required property.")
```
Comment (isacikgoz) [inline app/import_validators_test.go:936]:
```suggestion
		require.NotNil(t, err, "Should have failed due to missing required property.")
		assert.Equal(t, err.Id, "app.import.validate_post_import_data.user_missing.error")
```

File: app/import_validators_test.go [hunk 5]
```diff
@@ -895,109 +895,131 @@ func TestImportValidateReplyImportData(t *testing.T) {
...
+
+		data = PostImportData{
+			Team:     ptrStr("teamname"),
+			Channel:  ptrStr("channelname"),
+			User:     ptrStr("username"),
+			CreateAt: ptrInt64(model.GetMillis()),
+		}
+		err = validatePostImportData(&data, maxPostSize)
+		require.NotNil(t, err, "Should have failed due to missing required property.")
```
Comment (isacikgoz) [inline app/import_validators_test.go:945]:
```suggestion
		require.NotNil(t, err, "Should have failed due to missing required property.")
		assert.Equal(t, err.Id, "app.import.validate_post_import_data.message_missing.error")
```

File: app/import_validators_test.go [hunk 6]
```diff
@@ -895,109 +895,131 @@ func TestImportValidateReplyImportData(t *testing.T) {
...
+
+		data = PostImportData{
+			Team:    ptrStr("teamname"),
+			Channel: ptrStr("channelname"),
+			User:    ptrStr("username"),
+			Message: ptrStr("message"),
+		}
+		err = validatePostImportData(&data, maxPostSize)
+		require.NotNil(t, err, "Should have failed due to missing required property.")
```
Comment (isacikgoz) [inline app/import_validators_test.go:954]:
```suggestion
		require.NotNil(t, err, "Should have failed due to missing required property.")
		assert.Equal(t, err.Id, "app.import.validate_post_import_data.create_at_missing.error")
```

File: app/import_validators_test.go [hunk 7]
```diff
@@ -895,109 +895,131 @@ func TestImportValidateReplyImportData(t *testing.T) {
...
+	t.Run("Test with invalid message", func(t *testing.T) {
+		data := PostImportData{
+			Team:     ptrStr("teamname"),
+			Channel:  ptrStr("channelname"),
+			User:     ptrStr("username"),
+			Message:  ptrStr(strings.Repeat("0", maxPostSize+1)),
+			CreateAt: ptrInt64(model.GetMillis()),
+		}
+		err := validatePostImportData(&data, maxPostSize)
+		require.NotNil(t, err, "Should have failed due to too long message.")
```
Comment (isacikgoz) [inline app/import_validators_test.go:966]:
```suggestion
		require.NotNil(t, err, "Should have failed due to too long message.")
		assert.Equal(t, err.Id, "app.import.validate_post_import_data.message_length.error")
```

File: app/import_validators_test.go [hunk 8]
```diff
@@ -895,109 +895,131 @@ func TestImportValidateReplyImportData(t *testing.T) {
...
+	t.Run("Test with invalid CreateAt", func(t *testing.T) {
+		data := PostImportData{
+			Team:     ptrStr("teamname"),
+			Channel:  ptrStr("channelname"),
+			User:     ptrStr("username"),
+			Message:  ptrStr("message"),
+			CreateAt: ptrInt64(0),
+		}
+		err := validatePostImportData(&data, maxPostSize)
+		require.NotNil(t, err, "Should have failed due to 0 create-at value.")
```
Comment (isacikgoz) [inline app/import_validators_test.go:978]:
```suggestion
		require.NotNil(t, err, "Should have failed due to 0 create-at value.")
		assert.Equal(t, err.Id, "app.import.validate_post_import_data.create_at_zero.error")
```

File: app/import_validators_test.go [hunk 9]
```diff
@@ -895,109 +895,131 @@ func TestImportValidateReplyImportData(t *testing.T) {
...
+		data := PostImportData{
+			Team:     ptrStr("teamname"),
+			Channel:  ptrStr("channelname"),
+			User:     ptrStr("username"),
+			Message:  ptrStr("message"),
+			Props:    &props,
+			CreateAt: ptrInt64(model.GetMillis()),
+		}
+		err := validatePostImportData(&data, maxPostSize)
+		require.NotNil(t, err, "Should have failed due to long props.")
```
Comment (isacikgoz) [inline app/import_validators_test.go:1021]:
```suggestion
		require.NotNil(t, err, "Should have failed due to long props.")
		assert.Equal(t, err.Id, "app.import.validate_post_import_data.props_too_large.error")
```

**Comment by mgdelacroix** [comment]
Great suggestion @isacikgoz!! Added :relaxed:

File: app/import_functions.go (hunk 1 above)
Comment (agnivade) [reply app/import_functions.go:1084]: This is a very common mistake I am seeing in a lot of places. Wondering if a vet check can help here.

**Pull request merged**